src/modules/GitView/components/CommitModal.vue

```diff
@@ -0,0 +1,55 @@
+<template>
+  <div class="absolute top-20">
+    <div
+      class="border-black p-4 w-40 max-h-250px h-200px w-300px bg-red-500"
+      v-if="isOpen"
+      @click="away(isOpen)"
+    >
+      <p>{{ message }}</p>
+      <p>
+        Author : <b>{{ author }}</b>
+      </p>
+      <DsButton @click="$emit('onCheckout', sha)" class="mb-4 mt-4">
+        Checkout
+      </DsButton>
+    </div>
+  </div>
+</template>
+
+<script>
+import { computed } from '@vue/reactivity';
+
+export default {
+  name: 'CommitModal',
+
+  props: {
+    isOpen: {
+      type: Boolean,
+      default: false,
+    },
+    commit: {
+      type: Object,
+      required: true,
+    },
+  },
+
+  emits: ['onCheckout'],
+
+  setup(props) {
+    const away = (trigger) => {
```

Comment:
This wouldn't work since `isOpen` is a prop. You should use the v-model attribute and add an `emit` to set the `isOpen` prop